binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/test/java/org/springframework/cloud/stream/binder/kafka/streams/function/FunctionUnitDeriverTest.java

```diff
@@ -0,0 +1,24 @@
+package org.springframework.cloud.stream.binder.kafka.streams.function;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.util.StringUtils;
+
+import java.util.Arrays;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+class FunctionUnitDeriverTest {
+	@Test
+	void testFunctionNameDerivation() {
```

Comment:
Maybe a bit more descriptive, as the scenario under test is a function definition with explicit spaces in between?

Reply:
> Maybe a bit more descriptive, as the scenario under test is a function definition with explicit spaces in between?

Some of the functions have varied length spaces between them, two of them have none - the intention was just to show the fact they are fully trimmed regardless was always consistent, but I'll update the name of the test